core/task.go

```diff
@@ -295,21 +295,29 @@ func CreateTaskFromContent(body io.ReadCloser,
 	if tr.MaxCollectDuration != "" {
 		dl, err := time.ParseDuration(tr.MaxCollectDuration)
 		if err != nil {
-			return nil, err
+			return nil, serror.New(err)
 		}
 		opts = append(opts, SetMaxCollectDuration(dl))
 	}
 
 	if fp == nil {
-		return nil, errors.New("Missing workflow creation routine")
+		return nil, serror.New(errors.New("Missing workflow creation routine"))
 	}
 	task, errs := fp(sch, tr.Workflow, *mode, opts...)
 	if errs != nil && len(errs.Errors()) != 0 {
 		var errMsg string
+		var errorFields map[string]interface{}
 		for _, e := range errs.Errors() {
-			errMsg = errMsg + e.Error() + " -- "
+			errorFields = e.Fields()
```

Comment:
`errorFields` will be overwritten on each of iteration, so only Fields from the last errors will be visible. It should be managed in a different manner.